execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -319,6 +319,10 @@ where
                 error!("should not remove the system contract registry key");
                 Err(Error::RemoveKeyFailure(RemoveKeyFailure::PermissionDenied))
             }
+            Key::EraSummary => {
+                self.named_keys.remove(name);
+                Ok(())
```

Comment:
at some point we ought to | together the various self.named_keys.remove(name); Ok(()) arms.